lib/src/booleans/boolean_crosses.dart

```diff
@@ -0,0 +1,180 @@
+import 'package:turf/src/invariant.dart';
+
+import '../../helpers.dart';
+import '../line_intersect.dart';
+import '../polygon_to_line.dart';
+import 'boolean_point_in_polygon.dart';
+
+/// [booleanCrosses] returns [true] if the intersection results in a geometry whose
+/// dimension is one less than the maximum dimension of the two source geometries
+/// and the intersection set is interior to both source geometries.
+/// [booleanCsses] returns [true] for only [MultiPoint]/[Polygon], [MultiPoint]/[LineString],
+/// [LineString]/[LineString], [LineString]/[Polygon], and [LineString]/[MultiPolygon] comparisons.
+/// Other comparisons are not supported as they are outside the OpenGIS Simple
+/// [Feature]s spec and may give unexpected results.
+/// example:
+/// ```dart
+/// var line1 = LineString(coordinates: [
+///   Position.of([-2, 2]),
+///   Position.of([4, 2])
+/// ]);
+/// var line2 = LineString(coordinates: [
+///   Position.of([1, 1]),
+///   Position.of([1, 2]),
+///   Position.of([1, 3]),
+///   Position.of([1, 4])
+/// ]);
+/// var cross = booleanCrosses(line1, line2);
+/// //=true
+/// ```
+bool booleanCrosses(GeoJSONObject feature1, GeoJSONObject feature2) {
+  var geom1 = getGeom(feature1);
+  var geom2 = getGeom(feature2);
+
+  var exception = Exception("$geom2 is not supperted");
+  if (geom1 is MultiPoint) {
+    if (geom2 is LineString) {
+      return doMultiPointAndLineStringCross(geom1, geom2);
+    } else if (geom2 is Polygon) {
+      return doesMultiPointCrossPoly(geom1, geom2);
+    } else {
+      throw exception;
+    }
+  } else if (geom1 is LineString) {
+    if (geom2 is MultiPoint) {
+      // An inverse operation
+      return doMultiPointAndLineStringCross(geom2, geom1);
+    } else if (geom2 is LineString) {
+      return doLineStringsCross(geom1, geom2);
+    } else if (geom2 is Polygon) {
+      return doLineStringAndPolygonCross(geom1, geom2);
+    } else {
+      throw exception;
+    }
+  } else if (geom1 is Polygon) {
+    if (geom2 is MultiPoint) {
+      // An inverse operation
+      return doesMultiPointCrossPoly(geom2, geom1);
+    } else if (geom2 is LineString) {
+      // An inverse operation
+      return doLineStringAndPolygonCross(geom2, geom1);
+    } else {
+      throw exception;
+    }
+  } else {
+    throw exception;
+  }
+}
+
+bool doMultiPointAndLineStringCross(
```

Comment:
those helper functions could be private, right?

Reply:
btw is there a reason, why these are not private? Are they already shared between functions? Do they need to be exposed?